lang/cs/Org.Apache.REEF.Driver/Evaluator/EvaluatorRequest.cs

```diff
@@ -85,6 +95,12 @@ internal EvaluatorRequest(int number, int megaBytes, int core, string rack, stri
         [DataMember]
         public string RuntimeName { get; private set; }
 
+        [DataMember]
+        public IList<string> NodeNames { get; private set; }
```

Comment:
If it is pure C#  code, ImmutableList is not a bad option and we can get it from System.Collections.Immutable Nuget. Here we are actually quite constrained by the type selected as this type will be passed to C++ bridge code which has less options. For example, ImmutableList is not supported in C++.